chainParsers/pendulum.js

```diff
@@ -0,0 +1,161 @@
+
+    parserName = 'Pendulum';
+
+    //change [garPallet:garPallet] to the location where the asset registry is located.  ex: [assets:metadata]
+    garPallet = 'assetRegistry';
+    garStorage = 'metadata';
+
+    //change [xcGarPallet:xcGarStorage] to the location where the xc registry is located.  ex: [assetManager:assetIdType]
+    xcGarPallet = 'assetRegistry'
+    xcGarStorage = 'metadata'
+
+    /*
+    Not every parachain has published its xc Asset registry. But we
+    can still augment xcAsset registry by inferring.
+
+    To augment the xcAsset by parsing, please provide an array of xcm extrinsicIDs
+    containing the xcAsset asset you try to cover:
+
+    augment = {
+        'relaychain-paraID': [{
+            paraID: 'paraID',
+            extrinsicIDs: ['extrinsicID']
+        }]
+    }
+    */
+
+    /*
+    Parachain usually does not publish native asset in its own xc registry.
+    Allow team to polish xcRegistry using the following format:
+
+    manualRegistry = {
+        "relaychain-paraID": [{
+            asset: {
+                "Token": "currencyID"
+            },
+            xcmV1Standardized: [{"network":"relaychain"},{"parachain":paraID},{palletInstance/generalKey/generalIndex...}],
+        }]
+    }
+    */
+    augment = {}
+    manualRegistry = {}
+
+    isXcRegistryAvailable = true
+
+    //step 1: parse gar pallet, storage for parachain's asset registry
+    async fetchGar(chainkey) {
+        // implement your gar parsing function here.
+        await this.processPendulumGar(chainkey)
+    }
+
+    //step 2: parse xcGar pallet, storage for parachain's xc asset registry
+    async fetchXcGar(chainkey) {
+        if (!this.isXcRegistryAvailable) {
+            // skip if xcGar parser is unavailable
+            console.log(`[${chainkey}] ${this.parserName} xcGar NOT IMPLEMENTED - SKIP`)
+            return
+        }
+        // implement your xcGar parsing function here.
+        await this.processPendulumXcGar(chainkey)
+    }
+
+    //step 3: Optional augmentation by providing (a) a list xcm extrinsicIDs or (b) known xcmInteriorKeys-assets mapping
+    async fetchAugments(chainkey) {
+        //[Optional A] implement your augment parsing function here.
+        await this.processPendulumAugment(chainkey)
+        //[Optional B ] implement your manual registry here.
+        await this.processPendulumManualRegistry(chainkey)
+    }
+
+    // Implement Pendulum gar parsing function here
+    async processPendulumGar(chainkey) {
+        console.log(`[${chainkey}] ${this.parserName} custom GAR parser`)
+        //step 0: use fetchQuery to retrieve gar registry at the location [assets:garStorage]
+        let a = await super.fetchQuery(chainkey, this.garPallet, this.garStorage, 'GAR')
+        if (a) {
+            // step 1: use common Asset pallet parser func available at generic chainparser.
+            let assetList = this.processGarAssetPallet(chainkey, a)
```

Comment:
shouldn't we use `processGarTokensPallet` as of the definition at the end of [Section 4](https://github.com/pendulum-chain/xcm-global-registry/blob/main/chainParsers/README.md#step-4---select-asset-parser-for-fetchgar)?

Reply:
Okay, confusing, but that is a good reason.